src/python/pants/backend/python/tasks/python_tool_prep_base.py

```diff
@@ -99,14 +101,20 @@ def _build_tool_pex(self, tool_subsystem, interpreter, pex_path):
 
   def execute(self):
     tool_subsystem = self.tool_subsystem_cls.scoped_instance(self)
-    pex_name = tool_subsystem.options_scope
-    pex_path = os.path.join(self.workdir, self.fingerprint, '{}.pex'.format(pex_name))
+    specs_fingerprint = stable_json_sha1(tool_subsystem.get_requirement_specs())
+    pex_name_base = tool_subsystem.options_scope
+    full_pex_filename = '{}-{}.pex'.format(pex_name_base, specs_fingerprint)
+    pex_path = os.path.join(
+      get_pants_cachedir(),
+      'python-tools',
+      self.fingerprint,
+      full_pex_filename)
 
     interpreter_cache = PythonInterpreterCache.global_instance()
     interpreter = interpreter_cache.select_interpreter_for_targets([])
 
     if not os.path.exists(pex_path):
-      with self.context.new_workunit(name='create-{}-pex'.format(pex_name),
+      with self.context.new_workunit(name='create-{}-pex'.format(pex_name_base),
                                      labels=[WorkUnitLabel.PREP]):
         self._build_tool_pex(tool_subsystem=tool_subsystem,
                              interpreter=interpreter,
```

Comment:
The tool critically depends on the version of python it's built for and so that needs to be part of the cache key. It looks like this was a pre-existing bug, but it now becomes more critcial since there is no `clean-all` fix. Typically we've had a subdir formed from `str(interpreter.identity)`.

Reply:
The interpreter subdir sounds like the right approach, I will make this change!